File: Python/0204.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mapping = {')': '(', '}': '{', ']': '['}

        for char in s:
            if char in mapping:
                top_element = stack.pop() if stack else '#'
                if mapping[char] != top_element:
                    return False
            else:
                stack.append(char)

        return not stack
    
# 測試
solution = Solution()
print(solution.isValid("()"))          # 預期輸出: True
print(solution.isValid("()[]{}"))      # 預期輸出: True
print(solution.isValid("(]"))          # 預期輸出: False
print(solution.isValid("([)]"))        # 預期輸出: False
print(solution.isValid("{[]}"))        # 預期輸出: True

File: Ruby/0204.rb
def is_valid(s)
  stack = []
  mapping = { ')' => '(', '}' => '{', ']' => '[' }

  s.each_char do |char|
    if mapping.key?(char)
      top_element = stack.pop || '#'
      return false if mapping[char] != top_element
    else
      stack.push(char)
    end
  end

  stack.empty?
end

# 範例測試
puts is_valid("()")        # true
puts is_valid("()[]{}")    # true
puts is_valid("(]")        # false
